uint8 oType
uint8 srcBits
uint8 absValBits
uint8 dstBits

